server/issue.go

```diff
@@ -404,6 +404,53 @@ func (p *Plugin) GetCreateIssueMetadataForProjects(instanceID, mattermostUserID
 	})
 }
 
+func (p *Plugin) httpGetCommentVisibilityFields(w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodGet {
+		return http.StatusMethodNotAllowed,
+			errors.New("Request: " + r.Method + " is not allowed, must be GET")
+	}
+
+	mattermostUserID := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserID == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	instanceID := r.FormValue("instance_id")
+
+	client, _, connection, err := p.getClient(types.ID(instanceID), types.ID(mattermostUserID))
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	params := map[string]string{
+		"fieldValue": r.FormValue("fieldValue"),
+		"expand":     r.FormValue("expand"),
+		"accountId":  connection.AccountID,
+	}
+	results, err := client.SearchCommentVisibilityFields(params)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	if results == nil {
+		return http.StatusInternalServerError, errors.New("failed to return any results")
+	}
+
+	bb, err := json.Marshal(results)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to marshal response")
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	_, err = w.Write(bb)
```

Comment:
check this everywhere
```suggestion
	if _, err = w.Write(bb); err != nil {
```